datafusion/sqllogictest/src/engines/conversion.rs

```diff
@@ -101,5 +101,70 @@ pub(crate) fn decimal_to_str(value: Decimal) -> String {
 }
 
 pub(crate) fn big_decimal_to_str(value: BigDecimal) -> String {
-    value.round(12).normalized().to_string()
+    // Round the value to limit the number of decimal places
+    let value = value.round(12).normalized();
+    // Format the value to a string
+    format_big_decimal(value)
```

Comment:
we could use bigdecimal `to_plain_string` (see https://github.com/akubera/bigdecimal-rs/issues/135) -- but it's not released yet afaict